### PR TITLE
reference: add error codes

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? 

Add error codes in dev/reference.

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [x] This PR is to align with: [pingcap/docs-cn#2213](https://github.com/pingcap/docs-cn/pull/2213)
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [ ] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: dev/reference/error-codes.md [hunk 1]
```diff
@@ -19,6 +19,94 @@ TiDB is compatible with the error codes in MySQL, and in most cases returns the
...
+| 8104 | The plugin version does not match. |
+| 8105 | The plugin is repeatedly loaded. |
+| 8106 | The plugin defines a system variable whose name does not begin with the plugin name. |
+| 8107 | The loaded plugin does not specify a version, or the specified version is too low. |
+| 8108 | Unsupported execution plan type. |
+| 8109 | The specified index cannot be found when the index is analyzed. |
+| 8110 | The Cartesian product operation cannot be executed. | Set `cross-join` in the configuration to `true`. |
+| 8111 | When executing the `EXECUTE` statement, the corresponding `Prepare` statement cannot be found. |
+| 8112 | The number of parameters in the `EXECUTE` statement is not consistent with the `Prepare` statement. |
+| 8113 | The table schema involved in the `EXECUTE` statement has changed after the `Prepare` statement is executed. |
```
Comment (jackysp) [inline dev/reference/error-codes.md]:
```suggestion
| 8113 | The table schema related in the `EXECUTE` statement has changed after the `Prepare` statement is executed. |
```
`related` may be better.

File: dev/reference/error-codes.md [hunk 2]
```diff
@@ -19,6 +19,94 @@ TiDB is compatible with the error codes in MySQL, and in most cases returns the
...
+| 8102 | Unable to read the plugin definition information. |
+| 8103 | The plugin name is incorrect. |
+| 8104 | The plugin version does not match. |
+| 8105 | The plugin is repeatedly loaded. |
+| 8106 | The plugin defines a system variable whose name does not begin with the plugin name. |
+| 8107 | The loaded plugin does not specify a version, or the specified version is too low. |
+| 8108 | Unsupported execution plan type. |
+| 8109 | The specified index cannot be found when the index is analyzed. |
+| 8110 | The Cartesian product operation cannot be executed. | Set `cross-join` in the configuration to `true`. |
+| 8111 | When executing the `EXECUTE` statement, the corresponding `Prepare` statement cannot be found. |
```
Comment (jackysp) [inline dev/reference/error-codes.md]:
```suggestion
| 8111 | When executing the `EXECUTE` statement, the corresponding `Prepare` statement cannot be found.
```

File: dev/reference/error-codes.md [hunk 3]
```diff
@@ -19,6 +19,94 @@ TiDB is compatible with the error codes in MySQL, and in most cases returns the
...
+| 8101 | The format of plugin ID is incorrect. | The correct format is `[name]-[version]`, and no `-` is allowed in `name` and `version`. |
+| 8102 | Unable to read the plugin definition information. |
+| 8103 | The plugin name is incorrect. |
+| 8104 | The plugin version does not match. |
+| 8105 | The plugin is repeatedly loaded. |
+| 8106 | The plugin defines a system variable whose name does not begin with the plugin name. |
+| 8107 | The loaded plugin does not specify a version, or the specified version is too low. |
+| 8108 | Unsupported execution plan type. |
+| 8109 | The specified index cannot be found when the index is analyzed. |
+| 8110 | The Cartesian product operation cannot be executed. | Set `cross-join` in the configuration to `true`. |
```
Comment (jackysp) [inline dev/reference/error-codes.md]:
```suggestion
| 8110 | The Cartesian product operation cannot be executed. Set `cross-join` in the configuration to `true`. |
```

Comment (ran-huang) [reply dev/reference/error-codes.md]: The English table schema has a `Solution` column. Should we put solutions in that column?

Comment (jackysp) [reply dev/reference/error-codes.md]: ok

**Comment by ran-huang** [comment]
@lilin90 PTAL~

File: dev/reference/error-codes.md [hunk 4]
```diff
@@ -19,6 +19,94 @@ TiDB is compatible with the error codes in MySQL, and in most cases returns the
 | 8003 | If the data in a row is not consistent with the index when executing the `ADMIN CHECK TABLE` command, TiDB returns this error. |
 | 8004 | A single transaction is too large. | See [the error message `transaction too large`](/dev/faq/tidb.md#the-error-message-transaction-too-large-is-displayed) for the cause and solution.  |
 | 8005 | Transactions in TiDB encounter write conflicts. | See [the Troubleshoot section](/dev/faq/tidb.md#troubleshoot) for the cause and solution. |
+| 8018 | The plugin cannot be reloaded because it has not been loaded before. |
+| 8019 | The version of the plugin that is being reloaded is different from the previous version. Therefore, the plugin cannot be reloaded. |
+| 8020 | The table is locked. |
+| 8021 | The key does not exist. |
+| 8022 | The transaction commit fails. You may retry the process. |
```
Comment (lilin90) [inline dev/reference/error-codes.md]:
```suggestion
| 8022 | The transaction commit fails. You can retry the process. |
```

File: dev/reference/error-codes.md [hunk 5]
```diff
@@ -19,6 +19,94 @@ TiDB is compatible with the error codes in MySQL, and in most cases returns the
...
+| 8023 | An empty value is not allowed. |
+| 8024 | Illegal transactions. |
```
Comment (lilin90) [inline dev/reference/error-codes.md]:
```suggestion
| 8024 | Invalid transactions. |
```

Comment (lilin90) [reply dev/reference/error-codes.md]: Use "invalid" not "illegal" in this context. Please update all "illegal" in this PR. Thanks!

File: dev/reference/error-codes.md [hunk 6]
```diff
@@ -19,6 +19,94 @@ TiDB is compatible with the error codes in MySQL, and in most cases returns the
...
+| 8025 | The single Key-Value pair that are being written is too large. |
```
Comment (lilin90) [inline dev/reference/error-codes.md]:
```suggestion
| 8025 | The single Key-Value pair being written is too large. |
```

File: dev/reference/error-codes.md [hunk 7]
```diff
@@ -19,6 +19,94 @@ TiDB is compatible with the error codes in MySQL, and in most cases returns the
...
+| 8026 | The interface has no implementation. |
```
Comment (lilin90) [inline dev/reference/error-codes.md]:
```suggestion
| 8026 | The interface has not been implemented. |
```

File: dev/reference/error-codes.md [hunk 8]
```diff
@@ -19,6 +19,94 @@ TiDB is compatible with the error codes in MySQL, and in most cases returns the
...
+| 8028 | The table schema has changed. |
+| 8029 | Incorrect value. |
+| 8030 | After an unsigned positive integer is converted to a signed integer, it exceeds the maximum value and displays as a negative integer.|
+| 8031 | When being converted to an unsigned integer, a negative integer is converted to a positive integer. |
+| 8032 | Illegal `year` format. |
+| 8033 | Illegal `year` value. |
+| 8034 | Incorrect `datetime` value. |
+| 8036 | Illegal `time` format. |
+| 8037 | Illegal `week` format. |
+| 8038 | The field cannot obtain a default value. |
```
Comment (lilin90) [inline dev/reference/error-codes.md]:
```suggestion
| 8038 | The field fails to obtain the default value. |
```

File: dev/reference/error-codes.md [hunk 9]
```diff
@@ -19,6 +19,94 @@ TiDB is compatible with the error codes in MySQL, and in most cases returns the
...
+| 8039 | Index offset is out of range. |
+| 8042 | The state of table schema does not exist. |
+| 8043 | The state of column information does not exist. |
+| 8044 | The state of index does not exist. |
+| 8045 | Illegal table data. |
+| 8046 | The state of column information is invisible. |
+| 8047 | The value of the system variable is not supported. This error usually occurs in the alarm information when the user sets a variable value that is not supported in the database.
```
Comment (lilin90) [inline dev/reference/error-codes.md]:
```suggestion
| 8047 | The value of the system variable is not supported. This error usually occurs in the alarm information when the user sets a variable value that is not supported in the database. |
```

**Comment by ran-huang** [comment]
@lilin90 Thanks. All suggestions applied. PTAL~

**Pull request merged**